test/ds/test_dynamic_shape_models_dynamo.py

```diff
@@ -0,0 +1,125 @@
+import argparse
+import os
+import sys
+import random
+
+parser = argparse.ArgumentParser(add_help=False)
+parser.add_argument('--verbosity', type=int, default=2)
+FLAGS, leftovers = parser.parse_known_args()
+sys.argv = [sys.argv[0]] + leftovers
+
+import numpy as np
+import unittest
+import torch
+import torch._dynamo
+import torch_xla.core.xla_model as xm
+import torch_xla.debug.metrics as met
+
+# It enables us to run python implementations of CompositeAutogradImplicit ops.
+# CompositeAutogradImplicit means we don't have an explicit backward formula for an op instead an op is composed of a bunch of ops that do have backward formulas and combines this formulas is equivalent to differentiating the op explicitly.
+pd = torch._C._EnablePythonDispatcher()
+xla_dev = xm.xla_device()
+torch._dynamo.config.capture_dynamic_output_shape_ops = True
+
+
+class Feedforward(torch.nn.Module):
+
+  def __init__(self, input_size, hidden_size):
+    super().__init__()
+    self.input_size = input_size
+    self.hidden_size = hidden_size
+    self.fc1 = torch.nn.Linear(self.input_size, self.hidden_size)
+    self.fc1.weight.data.fill_(0.01)
+    self.fc1.bias.data.fill_(0.01)
+    self.relu = torch.nn.ReLU()
+    self.fc2 = torch.nn.Linear(self.hidden_size, 1)
+    self.fc2.weight.data.fill_(0.01)
+    self.fc2.bias.data.fill_(0.01)
+    self.sigmoid = torch.nn.Sigmoid()
+
+  def forward(self, x):
+    hidden = self.fc1(x)
+    relu = self.relu(hidden)
+    output = self.fc2(relu)
+    output = self.sigmoid(output)
+    return output
+
+
+class TestDynamicShapeModels(unittest.TestCase):
+  def test_benchmark_backwards_pass_with_dynamic_input(self):
+    num_features = 2
+    model = Feedforward(num_features, hidden_size=10)
+    model.compile()
+
+    for i in range(0, 100):
+        self.backwards_pass_with_dynamic_input(model, num_features)
+
+  def test_backward_pass_with_dynamic_input(self):
+    num_features = 2
+    model = Feedforward(num_features, hidden_size=10)
+    self.backwards_pass_with_dynamic_input(model, num_features)
+
+  def test_compiled_backward_pass_with_dynamic_input(self):
+    num_features = 2
+    model = Feedforward(num_features, hidden_size=10)
+    model.compile()
+    self.backwards_pass_with_dynamic_input(model, num_features)
+
+  def backwards_pass_with_dynamic_input(self, model, num_features):
```

Comment:
nit: mind making it private "_backwards_pass_with_dynamic_input"?

Reply:
Done.